TWI-48 Reply

pm888 opened the pull request:
Reply tweet with parent id

pm888 committed:
TWI-48 Reply

Reply tweet with parent id

bunin commented on openapi.yaml:
```diff
@@ -248,7 +248,40 @@ paths:
+            schema:
+              $ref: '#/components/schemas/ReplyRequest'
+      responses:
+        '201':
```
you do not return 201 code in the code

bunin commented on migrations/migrations.go:
```diff
@@ -79,6 +79,17 @@ var items = []darwin.Migration{
+	{
+		Version:     7,
+		Description: `reply`,
+		Script: `CREATE TABLE reply (
```
why not using a main table?

bunin commented on internal/tweets/tweetsStr.go:
```diff
@@ -34,3 +34,7 @@ type TweetValid struct {
 	ValidErr map[string]string
 }
+type TweetReply struct {
+	ReplyId int
```
this does not seem to be used/needed

pm888 replied:
done

bunin replied:
done where?

pm888 replied:
done

bunin reviewed (changes requested):
https://github.com/mo-pm888/twitter/pull/37/files#r1259883470

pm888 committed:
TWI-48

Add separate fun "CreatNewTweet"

bunin commented on internal/services/helpers.go:
```diff
@@ -134,3 +137,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(errj)
 }
+func CreatNewTweet(newTweet tweets.CreatNewTweet, r *http.Request, userID int, tweetTo string) error {
```
passing request around is not a good idea, if you only need a context pass a context

pm888 replied:
done

bunin commented on internal/services/helpers.go:
```diff
@@ -134,3 +137,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 }
+func CreatNewTweet(newTweet tweets.CreatNewTweet, r *http.Request, userID int, tweetTo string) error {
+	query := `INSERT INTO tweets (user_id, text, created_at, public, only_followers, only_mutual_followers, only_me,reply_to)
+		VALUES ($1, $2, $3, $4, $5, $6, $7,&8) RETURNING tweet_id`
```
`&8` ???

pm888 replied:
done

bunin commented on internal/users/users.go:
```diff
@@ -10,21 +10,42 @@ import (
+	realName bool
+}
+
+func (v *UserValid) Error() string {
```
this should not be here

bunin commented on internal/users/users.go:
```diff
@@ -309,47 +330,6 @@ func UnfollowUser(w http.ResponseWriter, r *http.Request) {
 	fmt.Println("Done")
 }
-
-func EditProfile(w http.ResponseWriter, r *http.Request) {
```
this should not be here

pm888 committed:
TWI-48

Fix bugs

pm888 committed:
Merge remote-tracking branch 'origin/main' into TWI-48

# Conflicts:
#	openapi.yaml

bunin commented on internal/services/helpers.go:
```diff
@@ -134,3 +138,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(errj)
 }
+func CreatNewTweet(newTweet tweets.CreatNewTweet, ctx context.Context, userID int, tweetTo string) error {
```
typo

pm888 committed:
TWI-48

Create a new reply
Add "create_tweet" function
Fix migration,createNewTweet,creatNewTweet struct,

bunin commented on internal/server/server.go:
```diff
@@ -30,9 +30,9 @@ func Server() {
-	r.HandleFunc("/v1/users", func(w http.ResponseWriter, r *http.Request) {
-		Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.EditProfile)).ServeHTTP(w, r)
-	}).Methods(http.MethodPatch)
+	//r.HandleFunc("/v1/users", func(w http.ResponseWriter, r *http.Request) {
+	//	Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.EditProfile)).ServeHTTP(w, r)
+	//}).Methods(http.MethodPatch)
```
why is it here?

bunin commented on internal/users/users.go:
```diff
@@ -547,3 +542,49 @@ func DeleteUserSession(token string) error {
 
 	return nil
 }
+func RegisterUsersValidations(userValid *UserValid) error {
```
this does not seem to have anything to do with Reply
